File: iOS/swiftui-journey-module/JourneyModuleSample/JourneyModuleSample/Callbacks/TermsAndConditionsCallbackView.swift
//
//  TermsAndConditionsCallbackView.swift
//  JourneyModuleSample
//
//  Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
//
//  This software may be modified and distributed under the terms
//  of the MIT license. See the LICENSE file for details.
//

import SwiftUI
import PingJourney
import Combine

/**
 * A SwiftUI view for presenting and accepting terms and conditions during authentication flows.
 *
 * This view displays the terms and conditions text along with version and creation date information.
 * Users must review the terms and toggle their acceptance before proceeding. This is commonly used
 * during registration or when terms are updated and require re-acceptance. The toggle reflects
 * the current acceptance state.
 *
 * **User Action Required:** YES - User must toggle the switch to accept the terms and conditions.
 *
 * The UI displays the version, creation date, full terms text, and an acceptance toggle switch.
 * All information is presented in a clear hierarchy with appropriate styling.
 */
struct TermsAndConditionsCallbackView: View {
    let callback: TermsAndConditionsCallback
    let onNodeUpdated: () -> Void

    @State var accepted: Bool = false

    var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            // Version
            if !callback.version.isEmpty {
                Text(callback.version)
                    .font(.headline)
                    .foregroundColor(.primary)
            }

            // Create Date
            if !callback.createDate.isEmpty {
                Text(callback.createDate)
                    .font(.headline)
                    .foregroundColor(.primary)
            }

            // Terms Text
            if !callback.terms.isEmpty {
                Text(callback.terms)
                    .font(.subheadline)
                    .foregroundColor(.secondary)
                    .multilineTextAlignment(.leading)
            }

            // Acceptance Toggle
            Toggle("I accept the terms and conditions", isOn: $accepted)
                .toggleStyle(SwitchToggleStyle())
                .onChange(of: accepted) { newValue in
                    callback.accepted = newValue
                }
        }
        .padding()
        .onAppear {
            accepted = callback.accepted
        }
    }
}

File: flutter/flutter-journey/lib/ui/features/journey/views/widgets/callbacks/terms_and_conditions_callback_view.dart
/*
 * Copyright (c) 2026 Ping Identity Corporation. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import 'package:flutter/material.dart';
import 'package:ping_journey/ping_journey.dart';

class TermsAndConditionsCallbackView extends StatelessWidget {
  const TermsAndConditionsCallbackView({
    super.key,
    required this.callback,
    required this.onChanged,
  });

  final TermsAndConditionsCallback callback;
  final VoidCallback onChanged;

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        if (callback.version.isNotEmpty) Text('Version: ${callback.version}'),
        if (callback.createDate.isNotEmpty)
          Text('Date: ${callback.createDate}'),
        ConstrainedBox(
          constraints: const BoxConstraints(maxHeight: 160),
          child: SingleChildScrollView(child: Text(callback.terms)),
        ),
        SwitchListTile(
          contentPadding: EdgeInsets.zero,
          title: const Text('I accept the terms and conditions'),
          value: callback.accepted,
          onChanged: (value) {
            callback.accepted = value;
            onChanged();
          },
        ),
      ],
    );
  }
}
